Wangscape/noise/module/ReseedableOps.cpp

```diff
@@ -0,0 +1,502 @@
+#include "ReseedableOps.h"
+#include "Exp.h"
+#include "Pow.h"
+#include "Gradient.h"
+#include "NormLPQ.h"
+#include "CornerCombinerBase.h"
+
+namespace noise
+{
+namespace module
+{
+
+ReseedablePtr operator+(ReseedablePtr left, ReseedablePtr right)
+{
+    auto add_p = std::make_shared<Reseedable<Add>>();
+    add_p->getModule().SetSourceModule(0, left->getModule());
+    add_p->getModule().SetSourceModule(1, right->getModule());
+
+    return makeModuleGroup({
+        {"source0", left},
+        {"source1", right},
+        {DEFAULT_MODULE_GROUP_OUT, add_p}
+    });
+}
+
+ReseedablePtr operator+(ReseedablePtr left, double right)
+{
+    auto scale_bias_p = std::make_shared<Reseedable<ScaleBias>>();
+    scale_bias_p->module.SetSourceModule(0, left->getModule());
+    scale_bias_p->module.SetScale(1.);
+    scale_bias_p->module.SetBias(right);
+
+    return makeModuleGroup({
+        {"source", left},
+        {DEFAULT_MODULE_GROUP_OUT, scale_bias_p}
+    });
+}
+
+ReseedablePtr operator+(double left, ReseedablePtr right)
+{
+    return right + left;
+}
+
+ReseedablePtr& operator+=(ReseedablePtr& left, ReseedablePtr right)
+{
+    return (left = left + right);
+}
+
+ReseedablePtr& operator+=(ReseedablePtr& left, double right)
+{
+    return (left = left + right);
+}
+
+ReseedablePtr operator-(ReseedablePtr left, ReseedablePtr right)
+{
+    return left + -right;
+}
+
+ReseedablePtr operator-(ReseedablePtr left, double right)
+{
+    return left + -right;
+}
+
+ReseedablePtr operator-(double left, ReseedablePtr right)
```

Comment:
That would only be possible for a `Reseedable` containing to a `Const` module, not the general case. In any case, all these operators will be deleted before this PR is merged.